src/data_generation.jl

```diff
@@ -0,0 +1,6 @@
+# Useful for LinearAlgebra tests
```

Comment:
This got its own file because I forsee that we will need a lot more of these to generate various kinds of structured matrixes, composites etc.